src/context.ts

```diff
@@ -179,7 +181,23 @@ export class NuxtIconModuleContext {
     const { getIconData } = await import('@iconify/utils')
     const { loadCollectionFromFS } = await import('@iconify/utils/lib/loader/fs')
 
+    const failed: string[] = []
+    let count = 0
+
+    const customCollectionNames = new Set(customCollections.map(c => c.prefix))
     const collections = new Map<string, IconifyJSON>()
+
+    function loadCollection(prefix: string) {
+      if (customCollectionNames.has(prefix)) {
+        const collection = customCollections.find(c => c.prefix === prefix)
+        if (collection) {
+          return Promise.resolve(collection)
```

Comment:
Using the `Promise.resolve` here to keep the return type `Promise<IconifyJSON | undefined>` and play nicely with existing logic of `iconifyCollectionMap`.